### PR TITLE
feat: spin up full mev

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
To run this do

`kurtosis run github.com/kurtosis-tech/eth2-package@gyani/full-mev '{"mev_type": "full"}'`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -41,7 +41,10 @@ jobs:
 
       - run: kurtosis engine restart
 
-      - run: kurtosis run ${PWD}
+      - run:
```
Comment (h4ck3rk3y) [inline .circleci/config.yml:44]: will revert post PR

File: src/mev_flood/mev_flood_launcher.star [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        config = ServiceConfig(
+            image = MEV_FLOOD_IMAGE,
+            entrypoint = ["/bin/sh", "-c", "touch main.log && tail -F main.log"]
+        )
+    )
+
+    plan.exec(
+        service_name = "mev-flood",
+        recipe = ExecRecipe(
+            command = ["/bin/sh", "-c", "./run init -r {0} -k 0xef5177cd0b6b21c87db5a0bf35d4084a8a57a9d6a064f86d51ac85f2b873a4e2 -s deployment.json".format(el_uri)]
```
Comment (h4ck3rk3y) [inline src/mev_flood/mev_flood_launcher.star:17]: pass user key too!! when the new image is out tomorrow

File: src/package_io/parse_input.star [hunk 3]
```diff
@@ -216,4 +217,26 @@ def enrich_mev_extra_params(parsed_arguments_dict, mev_prefix, mev_port):
 		if participant["cl_client_type"] == "prysm":
 			participant["validator_extra_params"].append("--enable-builder")
 			participant["beacon_extra_params"].append("--http-mev-relay={0}".format(mev_url))
+
+	num_participants = len(parsed_arguments_dict["participants"])
+
+	mev_url = "http://{0}{1}:{2}".format(mev_prefix, num_participants, mev_port)
+
+	mev_participant = {
```
Comment (h4ck3rk3y) [inline src/package_io/parse_input.star:225]: this should be behind a massive if condition for mev type

**Comment by kamilchodola** [comment]
Hi @h4ck3rk3y !

This is interesting PR and I'm checking it's capabilities and have a question:
Will tx-fuzzer also be included in transactions included for MEV? Or only MEV-flood will be responsible for creating txs included for MEV?

**Comment by kamilchodola** [comment]
Also I just started entire kurtosis with command from first message:
> kurtosis run github.com/kurtosis-tech/eth2-package@gyani/full-mev '{"mev_type": "full"}'
And in Flashbots boost relay I don't see any payloads delivered - shouldn't here I have all proposed blocks?

**Comment by h4ck3rk3y** [comment]
Hey @kamilchodola ! Thanks for your comments this is inspiring :) Wonder how you found Kurtosis.

`mev-flood` itself should be enough to create the required transactions but I am launching tx-fuzz just in case. Still hitting some edge cases with this hence this hasn't been merged yet.

https://github.com/flashbots/builder/issues/89 <- have an ongoing issue

Also discussing this with people from Flashbots & Ethereum in a separate discord. We discovered that I wasn't setting the `BUILDER_TX_SIGNING_KEY`; got a bit further but the work is still ongoing

**Comment by kamilchodola** [comment]
@h4ck3rk3y Thanks for those answers! Will be following those issues as well :)

I'm working in one of clients team and checking Kurtosis on how it works and exploring potential features :)

**Comment by h4ck3rk3y** [comment]
Closing in favor of #115

**Pull request closed without merging**